### ISSUE TITLE
Visibility Binding ignored on layout when using LayoutCompression

### ISSUE THREAD

**Issue by JKennedy24** [xamarin/Xamarin.Forms#1607]
### Description
I have a layout using layout compression as well as binding to Visibility as seen in the sample here: https://github.com/xamarin/xamarin-forms-samples/blob/master/UserInterface/LayoutCompression/LayoutCompression/Views/LayoutCompressionPage.xaml#L149

in my project like:
       ` <Grid x:Name="myGrid" CompressedLayout.IsHeadless="True" IsVisible="{Binding IsVisible}" Grid.Row="0">`

But changing the value of `IsVisible` does not change visibility of the grid

### Steps to Reproduce

1.  Load sample on Android (haven't tested on other platforms) [lct.zip](https://github.com/xamarin/Xamarin.Forms/files/1643094/lct.zip)

2.  click button on page `Change Visibility`
3.  See red grid still visible

### Expected Behavior
Red grid should appear and dissapear when changing the `IsVisible` property (clicking the button)

### Actual Behavior
Red grid is always visible

### Basic Information

- Version with issue: V2.5.0.1222

**Comment by StephaneDelcroix** [comment]
That's expected at this time.

by setting `IsHeadless="true"`, you're bypassing the usual layout code.

marking this as an enhancement

**Comment by JKennedy24** [comment]
@StephaneDelcroix Thanks for clarifying. In that case the `LayoutCompresion` sample is incorrect and probably needs rectifying, or a couple of pointers added to the documentation IMO

**Comment by StephaneDelcroix** [comment]
I tried to reproduce this with a ContentPage, as this
```
			var label = new Label { Text = "Label to hide" };
			var btn = new Button {
				Text = "hide the label",
				Command = new Command(() => label.IsVisible = !label.IsVisible)
			};
			var grid = new Grid {
				Children = {
					label
				}
			};
			Xamarin.Forms.CompressedLayout.SetIsHeadless (grid, true);
			Content = new StackLayout { Children = { grid, btn} };
```

and it works fine

**Comment by StephaneDelcroix** [comment]
oh, if I go a level deeper, like this
```csharp
			var label = new Label { Text = "Label to hide" };
			Grid grid = null;
			var btn = new Button {
				Text = "hide the grid",
				Command = new Command(() => grid.IsVisible = !grid.IsVisible)
			};
			grid = new Grid {
				Children = {
					label
				}
			};
			Xamarin.Forms.CompressedLayout.SetIsHeadless (grid, true);
			var stack = new StackLayout { Children = { grid, btn } };
			Xamarin.Forms.CompressedLayout.SetIsHeadless(stack, true);

			Content = stack;
```

the layout rearranges, but the grid and the label within doesn't disappear

**Comment by JKennedy24** [comment]
@StephaneDelcroix  Yes the second sample was what I was trying to achieve. The sample I was using has been fixed.

Here's the change set that fixed the sample I was following:

https://github.com/xamarin/xamarin-forms-samples/commit/114f10169e18f63e6be99edfbc530ac8991ecb15#diff-afd929730ec9964f4cdb22e786c51792
